core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -881,6 +884,16 @@ public void withStartupAttempts(int attempts) {
         this.startupAttempts = attempts;
     }
 
+    @Override
+    public void close() {
+        stop();
+    }
+
+    public SELF withCustomizer(Consumer<CreateContainerCmd> hook) {
```

Comment:
I've decided to go with "post" hooks only because GenericObject does "pre" part already, and this new API just allows you to configure the final result and override some configuration. Do you have any example where "pre" hook might be helpful? Because I don't :)